lessons/4.mdx

```diff
@@ -0,0 +1,818 @@
+---
+title: Testing your Tier NFT
+description:
+  Prove your Tier NFT contract works with automated tests! Builds on knowledge
+  from Lesson 3.
+icons: ['solidity', 'hardhat', 'openzeppelin']
+---
+
+# Lesson 4: Testing your Tier NFT
+
+In this lesson, we'll be adding automated tests for our smart contract created
+in Lesson 3.
+
+Here you'll:
+
+- Learn about the benefits of writing automated tests, general testing concepts
+  and best practices
+- Walk away with many practical tests for smart contracts
+- Nibble on some food for thought about test-driven development
+
+## Why automated testing?
+
+Testing can:
+
+- prove code actually works
+- help you write more modular code
+- ensure code doesn't break when you or others make changes or add new features.
+  People new to the code-base may not even know when something breaks without
+  tests.
+- ensure a bug is fixed and won't come back again
+- give you a safety net when you refactor code
+
+Tests are a tool for writing your code and making sure it keeps working as
+expected when changes are made.
+
+Automated tests don't fully replace manually testing your smart contract but
+it's an important tool to ensuring your contract will work as expected.
+
+## Let's add some tests
+
+### First add a test framework
+
+You've already setup hardhat in Lesson 2, which is a great framework for writing
+automated tests.
+
+Add a line in your `scripts` section of `package.json` so it looks like this
```

Comment:
between line 43 and here on 46, I have no idea what I am supposed to set up. I'm going on a bit of guess work. 

- We really need to make clear what the starting point is,..what tools we should be using. Do we go to an original project directory, cd into something, open it with our code editor,  or command line, and which commands do we run to get to *Add the line in your `scripts` section of `package json`.

Maybe say to follow the steps in lesson 2. 

Or a better alternative to make it a bit more exciting/rewarding would be to actually start of asking them 

"Do you remember how to do A in Lesson 2?, and when we got A going, we did B, When that gets going did you use X or Y to get Z? (you can give an answer or two on the way) Then you cd into C and run D. And you could close it of with. Now you should have A, B, C and D set up. If you need to check, go to Lesson 2 and" 

....then you give them the signposts to it.

Reply:
True, I assumed a lot and that they already went through Lesson 2, so like idea of adding signposts back to Lesson 2.

I may also move this into a `<ContentCallout />` section.

Reply:
I added more clarification in a callout.